ThunkLibs/libvulkan/Guest.cpp

```diff
@@ -36,18 +36,37 @@ const std::unordered_map<std::string_view, uintptr_t /* guest function address *
 #undef PAIR
     });
 
+// Fatally erroring function with a thunk-like interface. This is used as a placeholder for unknown Vulkan functions
+[[noreturn]] static void FatalError(void* raw_args) {
+    auto called_function = reinterpret_cast<PackedArguments<void, uintptr_t>*>(raw_args)->a0;
+    fprintf(stderr, "FATAL: Called unknown Vulkan function at address %p\n", reinterpret_cast<void*>(called_function));
+    __builtin_trap();
+}
+
+static PFN_vkVoidFunction MakeGuestCallable(const char* origin, PFN_vkVoidFunction func, const char* name) {
+    auto It = HostPtrInvokers.find(name);
+    if (It == HostPtrInvokers.end()) {
+        // If we can't infer the function signature, return a fatally erroring function in Debug builds.
+        // For Release builds, return nullptr in case the application has a fallback code path.
+        fprintf(stderr, "%s: Unknown Vulkan function at address %p: %s\n", origin, func, name);
+#ifdef NDEBUG
```

Comment:
I'd hide this behind a (thunk specific) #ifdef, not the generic NDEBUG, as it can affect application behaviour, and debug builds are expected to behave the same way